python/drned_xmnr/op/config_op.py

```diff
@@ -68,6 +68,24 @@ def perform(self):
                 str([self.state_filename_to_name(st) for st in state_files])}
 
 
+class ViewStateOp(ConfigOp):
+    action_name = 'view state'
+
+    def _init_params(self, params):
+        self.state_name = params.state_name
+
+    def perform(self):
+        self.log.debug("config_view_state() with device {0}".format(self.dev_name))
+        state_name = self.state_name
+        state_filename = self.state_name_to_filename(state_name)
+        if state_filename == []:
+            raise ActionError('no such state: ' + self.state_name)
+        with open(state_filename, 'r') as f:
+            state_str = f.read()
+            return {'success': state_str}
+        return {'failure': "Could not view " + state_name}
```

Comment:
I believe this last `return` statement is redundant, it will never be used.

Reply:
That can certainly happen, but in that case an exception is thrown and not caught until `Action:cb_action`, the `with` block does not stop it.

Reply:
I meant to say that open() might fail (not read()).

I can remove the last return if you insist but an error on open() will throw and exception. It's easy to test, just change the permission on the file so that current user can't read it (obviously not something we have to guard against but a file system error can give a similar result)

Reply:
I see the point now. But note that in that case any exception thrown by `open` would not be caught here either - the second `return` is never ever reached. You might want to wrap it all in a `try..except` block and return the failure from the exception handler.